SPACING.md

```diff
@@ -0,0 +1,335 @@
+
+```swift
+struct SomeResource: Resource {
+
+    static let path = "/resource"
+
+    var internalPropertyOne: String
+    var internalPropertyTwo: String
+
+    var coolDictionary: [String: String] {
+        return ["key": "value"]
+    }
+
+    lazy var something: Type = {
+        return Type(dict: coolDictionary)
+    }()
+
+    private var someNumber: Int
+    private var isSomething: Bool
+
+    /// Documentation
+    private var documentedProperty: String
+
+    /// Documentation
+    private var anotherDocumentedProperty: String
+}
+```
+
+### Property Observers & Getters/Setters
+- use __no__ blank lines at all between closures
+
+```swift
+var name: String {
+    willSet {
+        loadViewIfNeeded()
+    }
+    didSet {
+        nameLabel?.text = name
+    }
+}
+```
+
+### Overridden Functions
+- include __one__ blank line _after_ calling the super method if you add _multiple_ lines to your method
+- include __no__ blank line _after_ calling the super method if you only add _one_ line
+
+```swift
+override func viewWillAppear(_ animated: Bool) {
+    super.viewWillAppear(animated)
+
+    if let parent = parent {
+      // code
+    }    
+}
+
+override func viewDidAppear(_ animated: Bool) {
+    super.viewDidAppear(animated)
+    trackPageView()
+}
+```
+
+### Inside Protocol Definitions (and Enums or Structs without implementations)
+
+- __one__ blank line between the protocol definition and its first property/method, as well as between every property/method signature and (the documentation of) the following one
+- __one__ space between a property's type and every used item in `{ get set }`, `{ get }` or `{ set }`
+
+```swift
+protocol SomeDelegate: class {
+
+    /// Documentation
+    var property: Type { get set }
+
+    /// Documentation
+    func methodOne()
+
+    /// Documentation
+    func methodTwo()
+}
+
+struct Address {
+
+    /// Documentation
+    var street: String
+
+    /// Documentation
+    var city: String
+}
+```
+
+- if no documentation is required, use __no__ blank lines at all
+
+```swift
+enum Side {
+    case left
+    case right
+}
+```
+
+__Exception:__ Structs or enums declared _inside_ classes may occupy too much vertical space so that the blank lines could be omitted (however one might argue that such nested types should be declared in a separate extension anyway)
+
+```swift
+class BasketViewController: UIViewController {
+
+    fileprivate enum State {
+        /// An error occured while loading the basket.
+        case error
+        /// A loading operation that is related to the whole view is in progress.
+        case loading
+        /// The basket does not contain any products.
+        case emptyBasket
+        /// The basket contains products and should be displayed.
+        case filledBasket
+        /// There is currently no logged in user.
+        case notAuthenticated
+    }
+}
+```
+
+
+### Marks
+Marks should always be surrounded by __exactly one__ blank line above and below.
```

Comment:
Actually I'd suggest two blank lines above, and one blank line below the mark, to better convey the idea that the mark belongs thematically to the following block of code.

Reply:
**EXACTELY** That's what I've tried to suggest this for a long time, but most people somehow prefer one blank line which is why I added it this way. But now it's up for discussion again!

Reply:
Well, if you writing a style guide for the whole company I think it is acceptable to write "1 or 2 lines". Let those who insist on 1 or 2 specifically make the PR and explain the motivation behind that!